clients/stellar-relay-lib/src/node/mod.rs

```diff
@@ -19,6 +19,19 @@ pub struct NodeInfo {
 	pub network_id: NetworkId,
 }
 
+impl NodeInfo {
+	pub(crate) fn new(cfg: &NodeInfoCfg) -> Self {
```

Comment:
Let us create a new one and clone only the `version_str`, to avoid cloning the whole struct.